subsys/dfu/dfu_target/src/dfu_target_full_modem.c

```diff
@@ -30,21 +32,53 @@ int dfu_target_full_modem_cfg(const struct dfu_target_full_modem_params *params)
 		return -EALREADY;
 	}
 
-	if (params->buf == NULL || params->dev->dev == NULL) {
+	if (params->buf == NULL ||
+	   ((params->dev->dev == NULL) &&
+	    (!IS_ENABLED(CONFIG_DFU_TARGET_FULL_MODEM_USE_EXT_PARTITION)))) {
 		return -EINVAL;
 	}
 
-	/* Clone parametrs */
+	/* Clone parameters */
 	dfu_params.buf = params->buf;
 	dfu_params.len = params->len;
-	flash_dev = *params->dev;
 	dfu_params.dev = &flash_dev;
 
+#if defined(CONFIG_DFU_TARGET_FULL_MODEM_USE_EXT_PARTITION)
```

Comment:
Could use use `IS_ENABLED` here instead? and use a normal `if` instead of `#if`?

Reply:
i tried that initially, but it results in a build error due to the undefined flash area: `FLASH_AREA_ID(fmfu_storage)`.

Reply:
That makes sense.